### ISSUE TITLE
7.6.1 (18.c) Z should be finalized before Y

### ISSUE THREAD

**Issue by nholsti** [pat-rogers/Ada-202x-WG9-Informal-Review#147]
This says that "The goto statement will first cause Finalize(Y) to be called." However, RM 7.6.1 (11/3) says that objects are finalized in the reverse order of their creation, so Finalize(Z) should be called before Finalize(Y), because Z is declared (and created) after Y. The easiest fix is to reverse the order of the declarations of Y and Z in the example.
Note that RM 7.6.1 (19.a) also refers to this example and to Z.

**Comment by sttaft** [comment]
Another good catch!  Reversing the declaration order of Y and Z is likely to lead to other confusion.  I would suggest we systematically swap Y and Z references to be consistent with the finalization-is-in-reverse-order requirement.

**Comment by ARG-Editor** [comment]
Agreed. Applied Tucker's suggestion to the draft RM and placed it into AI12-0005-1 (the AARM AI).
